### PR TITLE
vp-603_lot_of_properties

### PR DESCRIPTION
**Pull request by AliveMen**
properties1

extend ui for edit properties blade

### CODE REVIEW

File: VirtoCommerce.CatalogModule.Web/Localizations/en.VirtoCommerce.Catalog.json [hunk 1]
```diff
@@ -199,7 +199,8 @@
       "property-list": {
         "subtitle": "Properties list",
         "labels": {
-          "empty-list": "None"
+          "empty-list": "None",
+          "modified": "Modified :"
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Localizations/en.VirtoCommerce.Catalog.json]: no need in space before colon in "Modified :".

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js [hunk 2]
```diff
@@ -14,11 +14,33 @@ angular.module('virtoCommerce.catalogModule')
...
+
+        $scope.resetFilter = function () {
+            blade.filtered = false;
+            _.each(blade.currentEntities,
+                function (prop) {
+                    prop.isSelected = true;
+                });
+        };
+
+        $scope.HasChangedProperties = function (properties) {
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: $scope.**h**asChangedProperties

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js [hunk 3]
```diff
@@ -105,8 +129,45 @@ angular.module('virtoCommerce.catalogModule')
 				canExecuteMethod: function () {
 					return true;
 				}
-			},
+            },
+            {
+                name: "Add filter", icon: 'fa fa-filter',
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: Use localization for the name

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js [hunk 4]
```diff
@@ -105,8 +129,45 @@ angular.module('virtoCommerce.catalogModule')
...
+                        }
+                    };
+                    bladeNavigationService.showBlade(newBlade, blade);
+                },
+                canExecuteMethod: function () {
+                    return true;
+                }
+            },
+            {
+                name: "clear filter", icon: 'fa fa-undo',
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: Use localization, and I think better to call Reset filter, as it is not clear all fields, but resets it to initial state with all fields selected.

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-selector.js [hunk 5]
```diff
@@ -0,0 +1,64 @@
+angular.module('virtoCommerce.catalogModule')
+    .controller('virtoCommerce.catalogModule.propertySelectorController', ['$scope', 'platformWebApp.bladeNavigationService', function ($scope, bladeNavigationService) {
+        var blade = $scope.blade;
+
+        blade.isLoading = true;
+
+        function initializeBlade() {
+            blade.title = 'Select properties for edit';
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-selector.js]: Please use resources for names.

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-selector.js [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        $scope.cancelChanges = function () {
+            bladeNavigationService.closeBlade(blade);
+        }
+
+        $scope.isValid = function () {
+            return _.some(blade.selectedEntities, function (item) { return item.length; })
+                || $scope.HasChangedProperties(blade.properties);
+        }
+
+        $scope.HasChangedProperties = function (properties) {
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-selector.js]: Need to remove capital H

Comment (yecli) [reply VirtoCommerce.CatalogModule.Web/Scripts/blades/property-selector.js]: Not fixed

Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Web/Scripts/blades/property-selector.js]: Done.

File: VirtoCommerce.CatalogModule.Web/Scripts/directives/property2.js [hunk 7]
```diff
@@ -62,9 +62,11 @@ angular.module('virtoCommerce.catalogModule').directive('vaProperty2', ['$compil
                     _.each(scope.currentEntity.values, function (x) { x.isInherited = false; });
 
                     ngModelController.$setViewValue(scope.currentEntity);
+                    scope.currentEntity.isChanged = true;
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/directives/property2.js]: I can't do this on this level

File: VirtoCommerce.CatalogModule.Web/Localizations/en.VirtoCommerce.Catalog.json [hunk 8]
```diff
@@ -751,8 +751,7 @@
           "select-categories": "Select categories…"
         }
       }
-    }
-  },
+    },
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Localizations/en.VirtoCommerce.Catalog.json]: Seems we get invalid json that way

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js [hunk 9]
```diff
@@ -27,7 +27,7 @@ angular.module('virtoCommerce.catalogModule')
             $scope.resetFilter();
 
             if ($localStorage.entryPropertyFilters) {
-                var savedFilteredProperties = $localStorage.entryPropertyFilters[authService.id];
+                var savedFilteredProperties = $localStorage.entryPropertyFilters[authService.id][blade.currentEntity.categoryId];
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: Thus we are changing the requirement that we are saving the filter for the user for any catalog.
I think we need to discuss it with Nikolay, as it was discussed before and proposed to save by categories, but we decided to choose one filter for the user.

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js [hunk 10]
```diff
@@ -188,15 +191,36 @@ angular.module('virtoCommerce.catalogModule')
                 name: "catalog.blades.property-list.labels.reset-filter", icon: 'fa fa-undo',
                 executeMethod: function () {
                     if ($localStorage.entryPropertyFilters) {
-                        $localStorage.entryPropertyFilters[authService.id] = [];
+
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: Probably removing `$localStorage.entryPropertyFilters[authService.id] = [];` could lead to errors if there were no filters saved.

File: VirtoCommerce.CatalogModule.Web/Scripts/directives/property2.js (hunk 7 above)
Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Web/Scripts/directives/property2.js]: removed

File: VirtoCommerce.CatalogModule.Web/Localizations/en.VirtoCommerce.Catalog.json (hunk 8 above)
Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Web/Localizations/en.VirtoCommerce.Catalog.json]: fixed

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js (hunk 9 above)
Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: Fixed

File: VirtoCommerce.CatalogModule.Data/packages.config [hunk 11]
```diff
@@ -5,7 +5,7 @@
   <package id="Common.Logging.Core" version="3.3.1" targetFramework="net461" />
   <package id="EntityFramework" version="6.2.0" targetFramework="net461" />
   <package id="FluentValidation" version="7.5.2" targetFramework="net461" />
-  <package id="Hangfire.Core" version="1.6.17" targetFramework="net461" />
+  <package id="Hangfire.Core" version="1.6.19" targetFramework="net461" />
```
Comment (yecli) [inline VirtoCommerce.CatalogModule.Data/packages.config]: Why do we need to update it?
Could you revert this change?

Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Data/packages.config]: reverted

**Comment by AliveMen** [comment]
"The property value deletion is not working now." - This bug happens only wirh BRAND property and conneted to default databese data. First time was found in properties bulk update implementation. Shoult be fixed outside of this PR as same as ugly UX with displaying culture names.

File: VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js [hunk 12]
```diff
@@ -14,14 +14,49 @@ angular.module('virtoCommerce.catalogModule')
 		function initialize(entity) {
 			blade.title = entity.name;
 			blade.subtitle = 'catalog.blades.property-list.subtitle';
-			blade.currentEntity = entity;
+            blade.currentEntity = entity;
```
Comment (tatarincev) [inline VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: The properties filtration based on user selection can be implemented by the simplest way (we have discussed with @AliveMen  abut details)

Comment (AliveMen) [reply VirtoCommerce.CatalogModule.Web/Scripts/blades/property-list.js]: done
